### PR TITLE
chore: Remove unused datasource query endpoint

### PR DESCRIPTION
**Pull request by nidhi-nair**
## Description
This PR removes an unused endpoint to reduce the scope of changes for multiple environments

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## Testing
No testing required for this change, regression would cover testing for this.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/DatasourceStructureSolutionCE.java [hunk 1]
```diff
@@ -16,7 +12,4 @@ Mono<DatasourceStructure> getStructure(
             Datasource datasource,
             boolean ignoreCache,
             String environmentName);
-
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/DatasourceStructureSolutionCE.java:19]: Should we also get rid of PluginExecutors getDatasourceMetadata method and it's overrides?

Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/DatasourceStructureSolutionCE.java:19]: You're right, thanks for catching that!

**Review by sondermanish — APPROVED** [review]
LGTM

**Pull request merged**